File: Java/PeopleWork/src/peoplework/Person.java
package peoplework;

/**
 * Programmer: Damian Zylski
 * Project:    PeopleWork
 * Class:      Person
 * Date:       08/10/2020
 * 
 */

public class Person
{
    //Attributes
    private String name;
    
    //Constructors
    public Person()
    {
        name = "Blank";
    }
    
    public Person(String s)
    {
        name = s;
    }
    
    //Setters
    public void setName(String s)
    {
        name = s;
    }
    //Getters
    public String getName()
    {
        return name;
    }
    //Output name
    public void writeOutput()
    {
        System.out.println("Name: " + name);
    }
    //check if Equal
    public boolean hasSameName(Person p)
    {
        if(this.name.equalsIgnoreCase(p.name))
        {
            return true;
        }
        else
        {
            return false;
        }
    }
    //compareTo
    public int compareTo(Person p)
    {
        if(this.name.compareToIgnoreCase(p.name) > 0)
        {
            return 1; //greater than
        }
        else if(this.name.compareToIgnoreCase(p.name) < 0)
        {
            return -1; //less than
        }
        else if(this.name.compareToIgnoreCase(p.name) == 0)
        {
            return 0; //equal
        }
        else
        {
            return 0; //default
        }
    }
    //toString
    public String toString()
    {
        return name;
    }
    
}

File: C++/SelectionSort/SelectionSort/Person.cpp
//Person Class implementation file
//Damian Zylski
//08/13/2020

#include "Person.h"
#include <iostream>
#include <string>
using namespace std;

//Constructors
Person::Person()
{
	name = "Blank";
}

Person::Person(string n)
{
	name = n;
}

//Destructor
Person::~Person()
{

}

//Getters
string Person::getName()
{
	return name;
}
//Setters
void Person::setName(string n)
{
	name = n;
}
//Write Output
void Person::writeOutput()
{
	cout << "Name: " << name << endl;
}
//Equal
bool Person::equal(Person p)
{
	if (this->name.compare(p.name) == 0)
	{
		return true;
	}
	else
	{
		return false;
	}
}
//Compare To
int Person::compareTo(Person p)
{
	if (this->name.compare(p.name) > 0)
	{
		return 1; //greater than
	}
	else if (this->name.compare(p.name) < 0)
	{
		return -1; //less than
	}
	else if (this->name.compare(p.name) == 0)
	{
		return 0; //equal
	}
	else
	{
		return 0; //default
	}
}